src/managers/itemTree.ts

```diff
@@ -257,15 +258,10 @@ export class ItemTreeManager extends ManagerTool {
   private async initializeGlobal() {
     const Zotero = this.getGlobal("Zotero");
     await Zotero.uiReadyPromise;
-    const window = this.getGlobal("window");
-    if (!Zotero._ItemTreeExtraColumnsGlobal) {
-      let globalCache = {
-        columns: [],
-        fieldHooks: {},
-        renderCellHooks: {},
-      } as ItemTreeExtraColumnsGlobal;
-      this.globalCache = Zotero._ItemTreeExtraColumnsGlobal = globalCache;
-
+    const window = this.getGlobal("window"),
+      globalCache = this.globalCache = ToolkitGlobal.getInstance(Zotero).itemTree;
+    if (!globalCache.initialized) {
+      globalCache.initialized = true;
```

Comment:
Possible confliction between multiple plugins.

Suppose A is just about to start initializing globalCache, B finds globalCache.initialized is true and thus try to access its props/methods, while they are not initialized yet.

Add a wait lock or load start & load finish flags for the initialization.

Reply:
I'm not sure the ready-idel-loding states is safe